service/src/main/java/uk/nhs/adaptors/gp2gp/gpc/builder/GpcRequestBuilder.java

```diff
@@ -108,16 +116,48 @@ private ParametersParameterComponent buildParamterComponent(String parameterName
 
     public RequestHeadersSpec<?> buildGetDocumentRecordRequest(GetGpcDocumentTaskDefinition documentTaskDefinition) {
         SslContext sslContext = requestBuilderService.buildSSLContext();
-        HttpClient httpClient = HttpClient.create().secure(t -> t.sslContext(sslContext));
+        HttpClient httpClient = buildHttpClient(sslContext);
         WebClient client = buildWebClient(httpClient);
 
         WebClient.RequestBodySpec uri = client
             .method(HttpMethod.GET)
-            .uri(gpcConfiguration.getDocumentEndpoint() + "/" + documentTaskDefinition.getDocumentId());
+            .uri(documentTaskDefinition.getAccessDocumentUrl());
 
         return buildRequestWithHeaders(uri, documentTaskDefinition, GPC_DOCUMENT_INTERACTION_ID);
     }
 
+    public RequestHeadersSpec<?> buildGetPatientIdentifierRequest(GetGpcDocumentReferencesTaskDefinition patientIdentifierTaskDefinition) {
+        SslContext sslContext = requestBuilderService.buildSSLContext();
+        HttpClient httpClient = buildHttpClient(sslContext);
+        WebClient client = buildWebClient(httpClient);
+
+        WebClient.RequestBodySpec uri = client
+            .method(HttpMethod.GET)
+            .uri(uriBuilder -> uriBuilder
```

Comment:
Could we move uri building code to another method, please?

Reply:
moved